x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlTranslatorHandler.java

```diff
@@ -56,4 +104,83 @@ public Query wrapFunctionQuery(ScalarFunction sf, Expression field, Supplier<Que
+            TypedAttribute attribute = checkIsPushableAttribute(bc.left());
+            Source source = bc.source();
+            Object value = ExpressionTranslators.valueOf(bc.right());
+            String format = null;
+            boolean isDateLiteralComparison = false;
+
+            // for a date constant comparison, we need to use a format for the date, to make sure that the format is the same
+            // no matter the timezone provided by the user
+            if (value instanceof ZonedDateTime || value instanceof OffsetTime) {
+                DateFormatter formatter;
+                if (value instanceof ZonedDateTime) {
+                    formatter = DateFormatter.forPattern(ExpressionTranslators.DATE_FORMAT);
+                    // RangeQueryBuilder accepts an Object as its parameter, but it will call .toString() on the ZonedDateTime instance
+                    // which can have a slightly different format depending on the ZoneId used to create the ZonedDateTime
+                    // Since RangeQueryBuilder can handle date as String as well, we'll format it as String and provide the format as well.
+                    value = formatter.format((ZonedDateTime) value);
+                } else {
+                    formatter = DateFormatter.forPattern(ExpressionTranslators.TIME_FORMAT);
+                    value = formatter.format((OffsetTime) value);
+                }
+                format = formatter.pattern();
+                isDateLiteralComparison = true;
+            } else if (attribute.dataType() == IP && value instanceof BytesRef bytesRef) {
+                value = DocValueFormat.IP.format(bytesRef);
+            } else if (attribute.dataType() == VERSION) {
+                // VersionStringFieldMapper#indexedValueForSearch() only accepts as input String or BytesRef with the String (i.e. not
+                // encoded) representation of the version as it'll do the encoding itself.
+                if (value instanceof BytesRef bytesRef) {
+                    value = new Version(bytesRef).toString();
+                } else if (value instanceof Version version) {
+                    value = version.toString();
+                }
+            } else if (attribute.dataType() == UNSIGNED_LONG && value instanceof Long ul) {
+                value = unsignedLongAsNumber(ul);
+            }
+
+            ZoneId zoneId = null;
+            if (DataTypes.isDateTime(attribute.dataType())) {
+                zoneId = bc.zoneId();
+            }
+
+            String name = pushableAttributeName(attribute);
+
+            Query query;
+            if (isDateLiteralComparison) {
+                // dates equality uses a range query because it's the one that has a "format" parameter
+                query = new RangeQuery(source, name, value, true, value, true, format, zoneId);
+            } else {
+                query = new TermQuery(source, name, value, DataTypes.isString(attribute.dataType()));
```

Comment:
I think `version` fields probably do want to push down case insensitive comparison.

Reply:
No version support for now (see above)